### ISSUE TITLE
XLSX crashes with autor issue because of illigal offset (array)

### ISSUE THREAD

**Issue by Leycarno** [PHPOffice/PhpSpreadsheet#2227]
This is:

```

- [ X] a bug report


### What is the expected behavior?

   read the excel proper and handle authorlist issues...

### What is the current behavior?

 "exception": [
        {
            "type": "TypeError",
            "code": 0,
            "message": "Illegal offset type",
            "file": "C:\\REPOS\\Eis20\\eis-pro-backend\\vendor\\phpoffice\\phpspreadsheet\\src\\PhpSpreadsheet\\Reader\\Xlsx.php",
            "line": 884
        }
    ]

### Solution /quickfix

**FILE : PhpOffice\PhpSpreadsheet\Reader\Xlsx.php
LINES 880 - 887**

  // Loop through contents
  foreach ($commentsFile->commentList->comment as $comment) {
      $commentModel = $docSheet->getComment((string) $comment['ref']);
      if (!empty($comment['authorId'] 
      **// !!! prove  for the existent key in the authors array
          && in_array($comment['authorId'], $authors))) {**
          $commentModel->setAuthor($authors[$comment['authorId']]);
      }
      $commentModel->setText($this->parseRichText($comment->text));
  }

**Comment by MarkBaker** [comment]
Duplicate of [Issue #2184](https://github.com/PHPOffice/PhpSpreadsheet/issues/2184)

**Comment by oleibman** [comment]
Closing. Fixed by PR #2329 which was merged in October 2021.
